experimental/table/server/master/src/main/java/alluxio/master/catalog/AlluxioCatalog.java

```diff
@@ -166,36 +164,14 @@ private Database getDatabaseByName(String dbName) throws NotFoundException {
    * @param constraint the column contraint
    * @return the partition info for the specified table
    */
-  public List<PartitionInfo> readTable(String dbName, String tableName,
+  public List<alluxio.grpc.catalog.Partition> readTable(String dbName, String tableName,
       Constraint constraint) throws IOException {
     Table table = getTable(dbName, tableName);
-    List<FieldSchema> cols = table.get().getPartitionKeys();
+    List<Partition> partitions = table.getPartitions();
 
-    List<PartitionInfo> parts = table.get().getPartitionInfo();
+    // TODO(david): implement partition pruning
```

Comment:
Are we going to need partition pruning in the future? since we are using hive connector now, I think it does filtering on the connector side..

Reply:
I don't know if it is necessary, but we may in the future.